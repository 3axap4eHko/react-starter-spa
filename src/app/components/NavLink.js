import React from 'react';
import withStyles from 'react-jss';
import classNames from 'classnames';
import createLink from 'react-steersman/createLink';

const styles = theme => ({
  root: {
    padding: 20,
  },
  activeRoot: {
    backgroundColor: theme.accentColor,
  },
  link: {
    color: theme.color,
  },
  activeLink: {},
});

const NavLink = createLink(({ to, title, classes, navigate, match }) => (
  <li className={classNames(classes.root, { [classes.activeRoot]: match })}>
    <a
      href={to}
      className={classNames(classes.link, { [classes.activeLink]: match })}
      title={title}
      onClick={(e) => {
        e.preventDefault();
        navigate(to);
      }}
    >
      {title}
    </a>
  </li>
));

export default withStyles(styles)(NavLink);
