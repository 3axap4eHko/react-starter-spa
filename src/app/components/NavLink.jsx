/* eslint-disable */

import React from 'react'
import { Route } from 'react-router'
import { Link } from 'react-router-dom'

const NavLink = ({ to, ...rest }) => (
  <Route
    path={typeof to === 'object' ? to.pathname : to}
    exact={true}
    strict={false}
    children={({ location, match }) => (
      <li className={match && 'active'}>
        <Link
          to={to}
          {...rest}
        />
      </li>
    )}
  />
);

export default NavLink