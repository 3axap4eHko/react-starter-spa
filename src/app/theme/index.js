export default {
  color: '#fff',
  backgroundColor: '#333',
  primaryColor: '#009688',
  accentColor: '#FF1744',
};
