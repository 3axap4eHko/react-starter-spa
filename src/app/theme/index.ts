export interface Theme {
  color: string
}

const theme: Theme = {
  color: '#212121'
};

export default theme;
