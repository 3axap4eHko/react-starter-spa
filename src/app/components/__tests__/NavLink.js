import React from 'react';
import renderer from 'react-test-renderer';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Steersman from 'react-steersman/Steersman';
import createMemoryHistory from 'react-steersman/createMemoryHistory';
import { ThemeProvider } from 'react-jss';
import NavLink from '../NavLink';
import theme from '../../theme';

configure({ adapter: new Adapter() });

test('Navigation Link with active styles', () => {
  const app = (
    <Steersman history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <NavLink to="/" title="Home" />
      </ThemeProvider>
    </Steersman>
  );
  const component = renderer.create(app);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = render(<div>{app}</div>);
  expect(wrapper.find('li[class*="Link-activeRoot"]').length).toBe(1);
  expect(wrapper.find('a[class*="Link-activeLink"]').length).toBe(1);
});

test('Navigation Link with styles', () => {
  const app = (
    <Steersman
      history={createMemoryHistory({
        initialEntries: ['/about'],
        initialIndex: 0,
      })}
    >
      <ThemeProvider theme={theme}>
        <NavLink to="/" title="Home" />
      </ThemeProvider>
    </Steersman>
  );
  const component = renderer.create(app);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = render(<div>{app}</div>);
  expect(wrapper.find('li[class*="Link-activeRoot"]').length).toBe(0);
  expect(wrapper.find('a[class*="Link-activeLink"]').length).toBe(0);
});

