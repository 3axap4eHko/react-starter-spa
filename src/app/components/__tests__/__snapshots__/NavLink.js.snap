// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Link with active styles 1`] = `
<li
  className="Link-root-0-1-1 Link-activeRoot-0-1-2"
>
  <a
    className="Link-link-0-1-3 Link-activeLink-0-1-4"
    href="/"
    onClick={[Function]}
    title="Home"
  >
    Home
  </a>
</li>
`;

exports[`Navigation Link with styles 1`] = `
<li
  className="Link-root-0-1-1"
>
  <a
    className="Link-link-0-1-3"
    href="/"
    onClick={[Function]}
    title="Home"
  >
    Home
  </a>
</li>
`;
